dcos_test_utils/gce.py

```diff
@@ -0,0 +1,174 @@
+            response = request.execute()
+            for instance_info in response['items']:
+                instance_info_list.append(instance_info)
+
+            request = self.deployment_manager.resources().list_next(previous_request=request, previous_response=response)
+
+        return instance_info_list
+
+    def get_deployment_info(self, deployment):
+        request = self.deployment_manager.deployments().get(project=self.project_id, deployment=deployment.name)
+        response = request.execute()
+        return response
+
+    def delete_deployment(self, deployment):
+        request = self.deployment_manager.deployments().delete(project=self.project_id, deployment=deployment.name)
+        response = request.execute()
+        return response
+
+    def set_metadata(self, deployment, body, instance_name):
+        request = self.compute.instances().setMetadata(project=self.project_id, zone=deployment.zone,
+                                                       instance=instance_name, body=body)
+        response = request.execute()
+        return response
+
+
+class Instance:
+    def __init__(self, gce_wrapper, name, priv_ip, pub_ip):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.priv_ip = priv_ip
+        self.pub_ip = pub_ip
+
+    def get_private_public_ip(self):
+        return {
+            'private_ip': self.priv_ip,
+            'public_ip': self.pub_ip}
+
+
+class Deployment:
+    def __init__(self, gce_wrapper, name, zone):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.zone = zone
+        self.instances = []
+        self.fingerprint = None
+
+    @catch_http_exceptions
+    def get_instances(self):
+        if not self.instances:
+            instance_info_list = self.gce_wrapper.list_deployment_instances(self)
+            for i in instance_info_list:
+                self.instances.append(Instance(self.gce_wrapper, i['name'],
+                                               i['networkInterfaces'][0]['networkIP'],
+                                               i['networkInterfaces'][0]['accessConfigs'][0]['natIP']))
+        return self.instances
+
+    @catch_http_exceptions
+    def get_fingerprint(self):
+        if not self.fingerprint:
+            self.fingerprint = self.get_info()['fingerprint']
+        return self.fingerprint
+
+    @catch_http_exceptions
+    def get_host_ips(self):
+        return [i.get_private_public_ip() for i in self.instances]
+
+    @catch_http_exceptions
+    def apply_ssh_key(self, public_key):
+        body = {
+            "kind": "compute#metadata",
+            'fingerprint': self.get_fingerprint(),
+            'items': [
+                {
+                    'key': 'ssh-keys',
+                    'value': 'dcos:ssh-rsa ' + public_key + ' dcos'
+                }
+            ]
+        }
+
+        for i in self.get_instances():
+            yield self.gce_wrapper.set_metadata(self, body, i['name'])
```

Comment:
I don't see where the value yielded here is used. Am I just blind? (Totally possible!)